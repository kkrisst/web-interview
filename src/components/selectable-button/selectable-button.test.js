import React from 'react'
import { shallow } from 'enzyme'

import SelectableButton from './selectable-button.component.jsx'

describe('Testing the SelectableButton component', () => {
  let wrapper
  beforeEach(() => {
    const props = {
      label: 'GP',
      selected: true,
      handleSelect: null,
    }
    wrapper = shallow(<SelectableButton {...props} />)
  })

  it('expect to render SelectableButton component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('has the class selected', () => {
    const selectedButton = wrapper.find('.selected')
    expect(selectedButton.length).toEqual(1)
  })

  it('has the label GP', () => {
    const selectedButton = wrapper.find('div')
    expect(selectedButton.text()).toEqual('GP')
  })

  it('can have other label than GP', () => {
    const props = {
      label: 'Therapist',
      selected: true,
      handleSelect: null,
    }
    const wrapper = shallow(<SelectableButton {...props} />)
    const selectedButton = wrapper.find('div')
    expect(selectedButton.text()).toEqual('Therapist')
  })

  it('expect handleSelect to be called', () => {
    const mockHandleSelect = jest.fn()

    const props = {
      label: 'GP',
      selected: true,
      handleSelect: mockHandleSelect,
    }
    const wrapper = shallow(<SelectableButton {...props} />)

    wrapper.find('div').simulate('click')
    expect(mockHandleSelect).toHaveBeenCalledTimes(1)
    wrapper.find('div').simulate('click')
    expect(mockHandleSelect).toHaveBeenCalledTimes(2)
  })
})
