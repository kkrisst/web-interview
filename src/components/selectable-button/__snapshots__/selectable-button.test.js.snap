// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the SelectableButton component expect to render SelectableButton component 1`] = `ShallowWrapper {}`;
